### PR TITLE
Ice for MATLAB linux support

### PR DESCRIPTION
**Pull request by pepone**
This PR adds Ice for MATLAB Linux support

We currently support MATLAB R2019b and MATLAB R2017b, see the README.md for the build requirements.

I have only tested with these MATLAB versions on Debian 10 and Ubuntu 20.04

### CODE REVIEW

File: matlab/README.md [hunk 1]
```diff
@@ -64,76 +69,96 @@ Use the following command to build the Ice toolbox package:
...
+### Prerequisites
 
-To use a source build, add the following directories to your MATLAB path:
- - `matlab\lib`
- - `matlab\lib\generated`
- - `matlab\lib\x64\Release`
+The build system requires MATLAB 2017b or MATLAB 2019b for Linux, with a supported C++ compiler and a Perl installation.
 
-#### Slice Files
+* For 2017b GCC 4.9 must be used, see [MATLAB 2017b supported compilers][2].
```
Comment (bernardnormier) [inline matlab/README.md:82]: Use GCC 4.9 for 2017b, see ...

File: matlab/README.md [hunk 2]
```diff
@@ -64,76 +69,96 @@ Use the following command to build the Ice toolbox package:
...
+* For 2019b GCC 6.3 must be used, see [MATLAB 2019b supported compilers][3].
```
Comment (bernardnormier) [inline matlab/README.md:83]: Same here.

File: matlab/src/Util.cpp [hunk 3]
```diff
@@ -59,10 +100,14 @@ IceMatlab::createStringFromUTF8(const string& s)
     }
     else
     {
-#ifdef _MSC_VER
-        //
+#if !defined(ICE_HAS_CODECVT_UTF8)
```
Comment (bernardnormier) [inline matlab/src/Util.cpp:103]: This `#if !defined` is not very elegant.

Could you rewrite it as:
```
#if defined(_MSC_VER)

#elif defined(ICE_HAS_CODECVT_UTF8)

#else

#endif
```
?

File: matlab/src/Util.cpp [hunk 4]
```diff
@@ -71,7 +116,9 @@ IceMatlab::createStringFromUTF8(const string& s)
         auto r = mxCreateCharArray(2, dims);
         auto buf = mxGetChars(r);
         int i = 0;
-#ifdef _MSC_VER
+#if !defined(ICE_HAS_CODECVT_UTF8)
```
Comment (bernardnormier) [inline matlab/src/Util.cpp:119]: See comment above.

**Review by bernardnormier — APPROVED** [review]
Looks good.

**Review by bentoi — APPROVED** [review]

**Review by InsertCreativityHere — APPROVED** [review]
Looks good to me!

**Pull request merged**